src/main/java/com/narmnevis/range/io/FilePublisher.java

```diff
@@ -53,4 +54,8 @@ public void close() throws IOException {
 		writer.close();
 	}
 
+	@Override
+	public String toString() {
+		return getClass().getName() + ":" + ((location !=null ) ? location.getAbsolutePath() : "no file");
```

Comment:
- There's a formatting space dropped in `location !=null` after `!=`. 
- Can change `no file` to a better wording as "unknown"  or "NA"?